### PR TITLE
Maintenance mode

### PR DESCRIPTION
**Pull request by fabclj**
This PR includes the InformationMessage component used for maintenance mode, Connectivity issue and Out of business hours views.
When enabled, is expected to have the information screen on chat open. HomeScreen will never be shown.

**How to test**
Append the following on settings object on your index.html
```
awaitEndpointConfig: true,
maintenance: {
    enabled: true,
    mode: "inform",
    text:" ",
    title: "",
}
```

US reference 55720

### CODE REVIEW

**Comment by kwinto** [comment]
One thing regrading Home Screen which comes to mind:

I would still expect to be able to access previous conversations even when your bussiness-chat is "closed". 
User may have some valuable info he is looking for. 

We can get the clarification on this to follow up, not blocking issue to me.

**Comment by fabclj** [comment]
> One thing regrading Home Screen which comes to mind:
> 
> I would still expect to be able to access previous conversations even when your bussiness-chat is "closed". User may have some valuable info he is looking for.
> 
> We can get the clarification on this to follow up, not blocking issue to me.

I agree, initially I did like that, but than I realized that we need more details. For example, the conversation should look exactly like expired one and the badge at the bottom shoulkd say something different, not just "Conversation ended", but something like "Conversation inactive". All buttons and interactions need to be disabled, etc.
Better to create a new US for this and define the requirements.

File: src/webchat-ui/components/WebchatUI.tsx [hunk 1]
```diff
@@ -685,8 +687,21 @@ export class WebchatUI extends React.PureComponent<React.HTMLProps<HTMLDivElemen
...
         const showRatingButton = enableRating && (enableRating === "always" || (enableRating === "once" && hasGivenRating === false));
+
+        let informMessage = "";
+        if (config.settings.maintenance.enabled && config.settings.maintenance.mode === "inform") {
+            informMessage = config.settings.maintenance.text || "This Webchat is disabled due to maintenance";
+        } else if (config.settings.connectivity.enabled && config.settings.connectivity.mode === "inform") {
+            informMessage = config.settings.connectivity.text || "This Webchat is disabled due to connectivity issues";
+        } else if (config.settings.businessHours.enabled && config.settings.businessHours.mode === "inform") {
+            informMessage = config.settings.businessHours.text || "This Webchat is disabled out of business hours";
+        }
```
[the remark this answers is not in the document]
Comment (kwinto) [reply src/webchat-ui/components/WebchatUI.tsx:698]: closing this, clarified

File: src/webchat-ui/components/presentational/InformationMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	gap: "24px",
+	maxWidth: "295px",
+	textAlign: "center",
+}));
+
+interface IInformationMessageProps {
+	message: string;
+}
+
+export const InformationMessage = (props: IInformationMessageProps) => {
```
[the remark this answers is not in the document]
Comment (kwinto) [reply src/webchat-ui/components/presentational/InformationMessage.tsx:29]: we can follow up on this, closing

**Review by kwinto — APPROVED** [review]

**Pull request merged**